File: SPCode/Reporting/Support/UI/SPReportFilterUI.vb
Namespace Reporting
    Public Class SPReportFilterUI

        Private _Filters As SPReportFilterCollection
        Private _Operators As ArrayList

        Public WriteOnly Property Filters() As SPReportFilterCollection
            Set(ByVal value As SPReportFilterCollection)
                _Filters = value
            End Set
        End Property

        Protected Overrides Sub OnLoad(ByVal e As System.EventArgs)
            If Not DesignMode Then
                FilterList.DataSource = SPReportUIFactory.GetDefaultFilterList
            End If
            BuildFieldList()
            BuildOperatorList()
            MyBase.OnLoad(e)
        End Sub

        Private Sub BuildFieldList()
            If _Filters IsNot Nothing Then
                bsFields.DataSource = _Filters

                Dim fieldColumn As DataGridViewComboBoxColumn = DirectCast(FilterList.Columns("FieldDataGridViewTextBoxColumn"), DataGridViewComboBoxColumn)

                If fieldColumn IsNot Nothing Then
                    With fieldColumn
                        .DataSource = bsFields
                        .DisplayMember = "DisplayMember"
                        .ValueMember = "ValueMember"
                    End With
                End If
            End If
        End Sub
        Private Sub BuildOperatorList()
            _Operators = SPComparisonsHelper.GetComparisonDataList(GetType(SPComparisons))
            If _Operators IsNot Nothing Then
                bsOperators.DataSource = _Operators

                Dim operatorColumn As DataGridViewComboBoxColumn = DirectCast(FilterList.Columns("OperationDataGridViewTextBoxColumn"), DataGridViewComboBoxColumn)

                If operatorColumn IsNot Nothing Then
                    With operatorColumn
                        .DataSource = bsOperators
                        .DisplayMember = "Name"
                        .ValueMember = "Value"
                    End With
                End If
            End If
        End Sub

        Private Sub btnAddFilter_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
            bsFilterTable.AddNew()
        End Sub

        Private Sub NewFilter_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles NewFilter.Click
            bsFilterTable.AddNew()
        End Sub

        Private Sub DeleteFilter_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DeleteFilter.Click
            Try
                bsFilterTable.RemoveCurrent()
            Catch ex As Exception
            End Try

        End Sub
    End Class
End Namespace

File: Backup/SoftLogic.Core/Win/Reporting/Support/UI/SPReportFilterUI.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;

namespace SoftLogik.Win
{
	namespace Reporting
	{
		public partial class SPReportFilterUI
		{
			
			
			private SPReportFilterCollection _Filters;
			private ArrayList _Operators;
			
			public SPReportFilterCollection Filters
			{
				set
				{
					_Filters = value;
				}
			}
			
			protected override void OnLoad(System.EventArgs e)
			{
				if (! DesignMode)
				{
					FilterList.DataSource = SPReportUIFactory.GetDefaultFilterList();
				}
				BuildFieldList();
				BuildOperatorList();
				base.OnLoad(e);
			}
			
			private void BuildFieldList()
			{
				if (_Filters != null)
				{
					bsFields.DataSource = _Filters;
					
					DataGridViewComboBoxColumn fieldColumn = (DataGridViewComboBoxColumn) (FilterList.Columns["FieldDataGridViewTextBoxColumn"]);
					
					if (fieldColumn != null)
					{
						fieldColumn.DataSource = bsFields;
						fieldColumn.DisplayMember = "DisplayMember";
						fieldColumn.ValueMember = "ValueMember";
					}
				}
			}
			private void BuildOperatorList()
			{
				_Operators = SPComparisonsHelper.GetComparisonDataList(typeof(SPComparisons));
				if (_Operators != null)
				{
					bsOperators.DataSource = _Operators;
					
					DataGridViewComboBoxColumn operatorColumn = (DataGridViewComboBoxColumn) (FilterList.Columns["OperationDataGridViewTextBoxColumn"]);
					
					if (operatorColumn != null)
					{
						operatorColumn.DataSource = bsOperators;
						operatorColumn.DisplayMember = "Name";
						operatorColumn.ValueMember = "Value";
					}
				}
			}
			
			private void btnAddFilter_Click(System.Object sender, System.EventArgs e)
			{
				bsFilterTable.AddNew();
			}
			
			public void NewFilter_Click(System.Object sender, System.EventArgs e)
			{
				bsFilterTable.AddNew();
			}
			
			public void DeleteFilter_Click(System.Object sender, System.EventArgs e)
			{
				try
				{
					bsFilterTable.RemoveCurrent();
				}
				catch (Exception)
				{
				}
				
			}
		}
	}
	
	
}
